internal/rbd/controllerserver.go

```diff
@@ -1664,6 +1664,8 @@ func (cs *ControllerServer) ControllerExpandVolume(
 
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+		// need update RequestedVolSize, because calcQosBasedOnCapacity use it.
+		rbdVol.RequestedVolSize = volSize
```

Comment:
I think there is a bug in this code here, where the resize was seccussful but later setting the QOS failed for some reason(container restart), when the next request comes, we will not be able to come inside this if as the `volSize == rbdVol.VolSize` as the resize was already done.

Reply:
updated